### PR TITLE
fix: Handle null-related test options

### PR DESCRIPTION
**Pull request by candiduslynx**
Follow-up for #1072 & #1002

### CODE REVIEW

**Comment by disq** [comment]
Same as https://github.com/cloudquery/plugin-sdk/pull/1074

**Comment by hermanschaaf** [comment]
This is the same as https://github.com/cloudquery/plugin-sdk/pull/1073 but has some more handling for ClickHouse cases that I couldn't test, so we can go with this one. Very similar either way :)

**Comment by hermanschaaf** [comment]
I get a nil pointer dereference error in `stripNullsFromLists` when I run this one against BigQuery destination though

**Comment by candiduslynx** [comment]
It also puts the sorting where it was missing in the inserts

**Comment by candiduslynx** [comment]
> I get a nil pointer dereference error in `stripNullsFromLists` when I run this one against BigQuery destination though

fixed in fff9fe5

**Review by hermanschaaf — APPROVED** [review]

File: plugin/nulls.go [hunk 1]
```diff
@@ -6,73 +6,75 @@ import (
...
+	return array.NewRecord(record.Schema(), cols, record.NumRows())
 }
 
 type AllowNullFunc func(arrow.DataType) bool
 
-// TODO(v4): use in v4
-//
-// nolint:unused
-func (f AllowNullFunc) replaceNullsByEmpty(records []arrow.Record) {
+func (f AllowNullFunc) replaceNullsByEmpty(record arrow.Record) arrow.Record {
```
Comment (yevgenypats) [inline plugin/nulls.go:48]: Is the method of a function necessary here?

Comment (candiduslynx) [reply plugin/nulls.go]: moved in 412d679eedeeefaa5a2c0308a6bbe230424abba7

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**